### PR TITLE
EA-2656: Improve handling of coref

### PR DESCRIPTION
**Pull request by ikattey**
Concepts now contain `skos:exactMatch`, while all other entity types use `owl:sameAs`

### CODE REVIEW

File: entity-management-web/src/main/java/eu/europeana/entitymanagement/web/service/EntityRecordService.java [hunk 1]
```diff
@@ -229,7 +229,7 @@ public EntityRecord createEntityFromRequest(EntityPreview entityCreationRequest,
 		 * sameAs will be replaced during consolidation; however we set this here to prevent duplicate
 		 * registrations if consolidation fails
 		 */
-		entity.setSameAs(Collections.singletonList(entityCreationRequest.getId()));
+		entity.setSameReferenceLinks(Collections.singletonList(entityCreationRequest.getId()));
```
Comment (gsergiu) [inline entity-management-web/src/main/java/eu/europeana/entitymanagement/web/service/EntityRecordService.java]: the sameAs is available in the request as well, probably it is better to merge that values as well

File: entity-management-web/src/main/java/eu/europeana/entitymanagement/web/service/EntityRecordService.java [hunk 2]
```diff
@@ -293,22 +293,22 @@ private String generateEntityId(String entityType, String entityId) {
...
-    public Optional<EntityRecord> retrieveCoreferenceSameAs(List<String> rdfResources) {
-	for (String resource : rdfResources) {
-	    Optional<EntityRecord> entityRecordOptional = retrieveByEntityId(resource);
-	    if (entityRecordOptional.isPresent()) {
-		return entityRecordOptional;
-	    }
-	}
+		public Optional<EntityRecord> retrieveCoreferencedEntity(List<String> rdfResources) {
+			for (String resource : rdfResources) {
+				Optional<EntityRecord> entityRecordOptional = retrieveByEntityId(resource);
```
Comment (gsergiu) [inline entity-management-web/src/main/java/eu/europeana/entitymanagement/web/service/EntityRecordService.java]: this method should send the database requests to search in sameAs/exactMatch fields. It currently tries to read by entityId. Additionally this method should send only one database request using $in operator

**Review by gsergiu — CHANGES_REQUESTED** [review]
it looks like we have a bug in the verification of co-references. I would suggest to fix them within this ticket

File: entity-management-web/src/main/java/eu/europeana/entitymanagement/web/service/EntityRecordService.java (hunk 1 above)
Comment (ikattey) [reply entity-management-web/src/main/java/eu/europeana/entitymanagement/web/service/EntityRecordService.java:232]: I don't understand this remark.

**Pull request merged**